dll/win32/shell32/folders/CDrivesFolder.cpp

```diff
@@ -57,6 +58,85 @@ static int iDriveTypeIds[7] = { IDS_DRIVE_FIXED,       /* DRIVE_UNKNOWN */
 *   IShellFolder implementation
 */
 
+static BOOL TryToLockDrive(HANDLE hDrive, DWORD dwRetryCount, DWORD dwSleep)
+{
+    DWORD dwError, dwBytesReturned;
+    for (DWORD i = 0; i < dwRetryCount; ++i)
+    {
+        if (DeviceIoControl(hDrive, FSCTL_LOCK_VOLUME, NULL, 0, NULL, 0, &dwBytesReturned, NULL))
+            return TRUE;
+
+        dwError = GetLastError();
+        if (dwError == ERROR_INVALID_FUNCTION)
+            break; /* don't sleep if function is not implemented */
+
+        Sleep(dwSleep);
+    }
+    SetLastError(dwError);
+    return FALSE;
+}
+
+// NOTE: See also https://support.microsoft.com/en-us/help/165721/how-to-ejecting-removable-media-in-windows-nt-windows-2000-windows-xp
+static BOOL DoEjectDrive(const WCHAR *physical, UINT nDriveType, INT& nStringID)
+{
+    DWORD dwAccessMode, dwShareMode = FILE_SHARE_READ | FILE_SHARE_WRITE;
+    if (nDriveType == DRIVE_REMOVABLE)
+        dwAccessMode = GENERIC_READ | GENERIC_WRITE;
```

Comment:
You shouldn't need GENERIC_WRITE for an umount, even on a RW volume. So, that if/else shouldn't exist. Just use GENERIC_READ.

Reply:
Ok.